### PR TITLE
sw-routing unit tests

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @gauntface

Some unit tests for `sw-routing`.

I'll follow up with end-to-end tests and unit + end-to-end tests for the other packages, but I wanted to keep the PRs small, and make sure I wasn't doing anything wrong w.r.t. the current testing best practices.

### CODE REVIEW

File: packages/sw-routing/src/lib/express-route.js [hunk 1]
```diff
@@ -40,7 +40,6 @@ class ExpressRoute extends Route {
    */
   constructor({path, handler, method}) {
     assert.isType({path}, 'string');
-    assert.hasMethod({handler}, 'handle');
```
Comment (gauntface) [inline packages/sw-routing/src/lib/express-route.js]: Why is this check being dropped?

File: packages/sw-routing/test/browser-unit/index.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        ui: 'bdd',
+        timeout: 10000
+      });
+
+      chai.should();
+      const expect = chai.expect;
+    </script>
+
+    <!-- In browser test scripts should be added to the page here
+    <script src="data/test-data.js"></script>
```
Comment (gauntface) [inline packages/sw-routing/test/browser-unit/index.html]: Nit: can these be removed?

File: packages/sw-routing/test/browser-unit/sw-unit/express-route.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    const namedParameterMatchingUrl = new URL('/value1/dummy/value2', location);
+    const namedParameterNonMatchingUrl = new URL('/value1/value2', location);
+
+    const route = new goog.routing.ExpressRoute({
+      handler, path: namedParameterPath,
+    });
+
+    const match = route.match({url: namedParameterMatchingUrl});
+    expect(Object.keys(match).length).to.equal(2);
+    expect(match.param1).to.equal('value1');
```
Comment (gauntface) [inline packages/sw-routing/test/browser-unit/sw-unit/express-route.js]: Nit: Move 'value1' to a variable and reuse. Will just be more obvious for someone else reading this in the future.

File: packages/sw-routing/test/browser-unit/sw-unit/regexp-route.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const captureGroupMatchingUrl = new URL('/value1/dummy/value2', location);
+    const captureGroupNonMatchingUrl = new URL('/value1/value2', location);
+
+    const route = new goog.routing.RegExpRoute({
+      handler, regExp: captureGroupRegExp,
+    });
+
+    const match = route.match({url: captureGroupMatchingUrl});
+    expect(match.length).to.equal(2);
+    expect(match[0]).to.equal('value1');
```
Comment (gauntface) [inline packages/sw-routing/test/browser-unit/sw-unit/regexp-route.js]: Nit: same as above - move to 'value1' to shared variable.

File: packages/sw-routing/test/browser-unit/sw-unit/route.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  it(`should not throw when Route() is called with valid handler and match parameters`, () => {
+    expect(() => new goog.routing.Route({handler, match})).not.to.throw();
+  });
+
+  it(`should throw when Route() is called with an invalid method`, () => {
+    expect(() => new goog.routing.Route({handler, match, method: invalidMethod})).to.throw();
+  });
+
+  it(`should not throw when Route() is called with a valid method`, () => {
+    expect(() => new goog.routing.Route({handler, match, method})).not.to.throw();
```
Comment (gauntface) [inline packages/sw-routing/test/browser-unit/sw-unit/route.js]: Nit: worth checking the method value has stuck?

File: packages/sw-routing/src/lib/express-route.js (hunk 1 above)
Comment (jeffposnick) [inline packages/sw-routing/src/lib/express-route.js]: In the course of considering whether I should write a test for it, I realized that the same assertion was being made in the constructor of the class it inherits from.

So it's not related to testing, but just something I snuck in.

File: packages/sw-routing/test/browser-unit/index.html (hunk 2 above)
Comment (jeffposnick) [inline packages/sw-routing/test/browser-unit/index.html]: Done.

File: packages/sw-routing/test/browser-unit/sw-unit/express-route.js (hunk 3 above)
Comment (jeffposnick) [inline packages/sw-routing/test/browser-unit/sw-unit/express-route.js]: Done.

File: packages/sw-routing/test/browser-unit/sw-unit/regexp-route.js (hunk 4 above)
Comment (jeffposnick) [inline packages/sw-routing/test/browser-unit/sw-unit/regexp-route.js]: Done.

File: packages/sw-routing/test/browser-unit/sw-unit/route.js (hunk 5 above)
Comment (jeffposnick) [inline packages/sw-routing/test/browser-unit/sw-unit/route.js]: Done. (I think I did what you're suggesting, at least.)

File: packages/sw-routing/src/lib/express-route.js (hunk 1 above)
Comment (gauntface) [inline packages/sw-routing/src/lib/express-route.js]: Cool - just wanted to check.

**Comment by gauntface** [comment]
LGTM.

**Pull request merged**